packages/fuels-test-helpers/src/script.rs

```diff
@@ -1,65 +1,21 @@
-#[cfg(feature = "fuel-core-lib")]
-use fuel_core::service::{Config, FuelService};
-
-#[cfg(not(feature = "fuel-core-lib"))]
-use crate::node::{Config, FuelService};
-
-use fuels_signers::provider::Provider;
-
-use fuel_gql_client::fuel_tx::{Receipt, Transaction};
-use fuels_contract::script::Script;
-use fuels_core::parameters::TxParameters;
-use fuels_types::errors::Error;
-
-/// Run the script binary located at `binary_filepath` and return its resulting receipts,
-/// without having to setup a node or contract bindings.
-pub async fn run_compiled_script(
-    binary_filepath: &str,
-    tx_params: TxParameters,
-    provider: Option<Provider>,
-) -> Result<Vec<Receipt>, Error> {
-    let script_binary = std::fs::read(binary_filepath)?;
-    let server = FuelService::new_node(Config::local_node()).await.unwrap();
-    let provider = provider.unwrap_or(Provider::connect(server.bound_address.to_string()).await?);
-
-    let script = build_script(script_binary, tx_params);
-
-    script.call(&provider).await
-}
-
-fn build_script(script_binary: Vec<u8>, tx_params: TxParameters) -> Script {
-    let tx = Transaction::Script {
-        gas_price: tx_params.gas_price,
-        gas_limit: tx_params.gas_limit,
-        maturity: tx_params.maturity,
-        receipts_root: Default::default(),
-        script: script_binary, // Pass the compiled script into the tx
-        script_data: vec![],
-        inputs: vec![],
-        outputs: vec![],
-        witnesses: vec![vec![].into()],
-        metadata: None,
-    };
-
-    Script::new(tx)
-}
-
 #[cfg(test)]
 mod tests {
-    use crate::script::run_compiled_script;
     use crate::{
         setup_single_asset_coins, setup_test_provider, DEFAULT_COIN_AMOUNT, DEFAULT_NUM_COINS,
     };
+    use fuels_contract::script::run_script_binary;
     use fuels_core::constants::BASE_ASSET_ID;
-    use fuels_core::parameters::TxParameters;
     use fuels_signers::WalletUnlocked;
     use fuels_types::errors::Error;
 
     #[tokio::test]
     async fn test_run_compiled_script() -> Result<(), Error> {
         // ANCHOR: run_compiled_script
         let path_to_bin = "../fuels/tests/logs/logging/out/debug/logging.bin";
-        let return_val = run_compiled_script(path_to_bin, TxParameters::default(), None).await?;
+        // TODO: use default provider
+        let (provider, _) = setup_test_provider(vec![], vec![], None).await;
+        // Provide `None` to all 3 other arguments so the function uses the default for each
+        let return_val = run_script_binary(path_to_bin, None, Some(provider), None).await?;
```

Comment:
Might be cleaner to just move `run_script_binary` to be a provider method or expand on our `Script` type and provide an interface that is similar to how we handle contracts. Just some ideas for the refactoring.